tinygrad/ops.py

```diff
@@ -229,7 +229,7 @@ def __init__(self, op:Optional[Union[UOps, Set[UOps]]]=None, arg:Any=None, src:O
                name:Optional[str]=None, dtype:Optional[Union[DType, Set[DType]]]=None, allow_any_len:bool=False, location=None,
                custom_early_reject:Optional[Set[Tuple[UOps, Any]]]=None):
     self.op: Optional[Tuple[UOps, ...]] = None if op is None else (tuple(op) if isinstance(op, set) else (op,))
-    self.dtype: Optional[Tuple[DType, ...]] = None if dtype is None else (tuple(dtype) if isinstance(dtype, set) else (dtype,))
+    self.dtype: Optional[Tuple[DType, ...]] = None if dtype is None or dtype is dtypes.void else (tuple(dtype) if isinstance(dtype, set) else (dtype,)) # noqa: E501
```

Comment:
`if dtype in (None, dtypes.void)` - but can dtype still be None?

Reply:
I like these suggestions, I'll change them. It appears that it can be None here, although it is possible this is the the result of tests instantiating certain values to None, I think I've found most of those occurrences though.